pkg/api-server/resource_endpoints.go

```diff
@@ -532,17 +532,54 @@ func (r *resourceEndpoints) validateLabels(rm model.ResourceMeta) validators.Val
 		}
 	}
 
-	for _, k := range maps.SortedKeys(rm.GetLabels()) {
+	if r.mode != config_core.Global {
+		zoneTag, ok := resource.GetMeta().GetLabels()[mesh_proto.ZoneTag]
+		if ok && zoneTag != r.zoneName {
+			err.AddViolationAt(validators.Root().Key(mesh_proto.ZoneTag), fmt.Sprintf("%s label should have %s value", mesh_proto.ZoneTag, r.zoneName))
+		}
+	}
+
+	r.validatePolicyRole(resource)
```

Comment:
please validate `r.descriptor` to have `.IsPolicy` and `.IsPluginOriginated`, you can avoid handling cast errors in this case

Reply:
added